### ISSUE TITLE
error on hello-word-go sample

### ISSUE THREAD

**Issue by lizzzcai** [OpenFunction/samples#16]
Hi, I like the design of OpenFunction and want to explore it on my end. However, I am facing issues when running the [sample](https://github.com/OpenFunction/samples/tree/main/functions/Knative/hello-world-go).

1. environment: minikube, Kubernetes v1.20.9
2. follow steps in docs to install the dependencies

```
❯ k get functions
NAME              BUILDSTATE   SERVINGSTATE   BUILDER                         SERVING   AGE
function-sample   Failed                      function-sample-builder-kfgfc             4m37s
❯ k get taskrun
NAME                                                 SUCCEEDED   REASON           STARTTIME   COMPLETIONTIME
function-sample-builder-kfgfc-buildrun-7vwxm-zz7cb   False       CouldntGetTask   4m38s       4m38s
```
below is the error in the TaskRun.
```
completionTime: '2021-09-23T17:41:58Z'
conditions:
  - lastTransitionTime: '2021-09-23T17:41:58Z'
    message: >-
      failed to create task run pod
      "function-sample-builder-kfgfc-buildrun-7vwxm-zz7cb": admission webhook
      "sidecar-injector.dapr.io" denied the request: service account
      'system:serviceaccount:tekton-pipelines:tekton-pipelines-controller' not
      on the list of allowed controller accounts. Maybe invalid TaskSpec
    reason: CouldntGetTask
    status: 'False'
    type: Succeeded
podName: ''
startTime: '2021-09-23T17:41:58Z'
taskSpec:
  params:
    - default: Dockerfile
      description: Path to the Dockerfile
      name: DOCKERFILE
      type: string
    - default: .
      description: The root of the code
      name: CONTEXT_DIR
      type: string
    - description: The URL of the image that the build produces
      name: shp-output-image
      type: string
    - description: The context directory inside the source directory
      name: shp-source-context
      type: string
    - description: The source directory
      name: shp-source-root
      type: string
    - default: 'openfunction/builder:v1'
      description: Image containing the build tools/logic
      name: BUILDER_IMAGE
      type: string
    - description: The name of where to store the app image.
      name: APP_IMAGE
      type: string
    - default: ''
      description: A subpath within the source input where the source to build is located.
      name: SOURCE_SUBPATH
      type: string
    - default: web
      description: The default process type to set on the image.
      name: PROCESS_TYPE
      type: string
    - default: ''
      description: Reference to a run image to use.
      name: RUN_IMAGE
      type: string
    - default: ''
      description: >-
        The name of the persistent app cache image (if no cache workspace is
        provided).
      name: CACHE_IMAGE
      type: string
    - default: 'false'
      description: Do not write layer metadata or restore cached layers.
      name: SKIP_RESTORE
      type: string
    - default: '1000'
      description: The user ID of the builder image user.
      name: USER_ID
      type: string
    - default: '1000'
      description: The group ID of the builder image user.
      name: GROUP_ID
      type: string
    - default: empty-dir
      description: The name of the platform directory.
      name: PLATFORM_DIR
      type: string
    - default: ''
      description: >-
        Environment variables to set during _build-time_. The formate is
        `key1=value1,key2=value2`.
      name: ENV_VARS
      type: string
  results:
    - description: The digest of the image
      name: shp-image-digest
    - description: The compressed size of the image
      name: shp-image-size
    - description: The commit SHA of the cloned source.
      name: shp-source-default-commit-sha
  steps:
    - args:
        - '--url'
        - 'https://github.com/OpenFunction/samples.git'
        - '--target'
        - $(params.shp-source-root)
        - '--result-file-commit-sha'
        - $(results.shp-source-default-commit-sha.path)
      command:
        - /ko-app/git
      image: >-
        quay.io/shipwright/git:nightly-2021-08-07-1628312894@sha256:d0ba59e6d3b8407ae28147c55f6e04eaf0883c8ad38ee5b75a114963265e935e
      name: source-default
      resources: {}
      securityContext:
        runAsGroup: 1000
        runAsUser: 1000
    - args:
        - '-c'
        - |
          #!/usr/bin/env bash
          set -e

          for path in "/cache" "/tekton/home" "/layers" "/workspace/source"; do
            echo "> Setting permissions on '$path'..."
            chown -R "$(params.USER_ID):$(params.GROUP_ID)" "$path"
          done

          echo "> Processing any environment variables..."
          ENV_DIR="/platform/env"

          echo "--> Creating 'env' directory: $ENV_DIR"
          mkdir -p "$ENV_DIR"

          envs=(`echo $(params.ENV_VARS) | tr ',' ' '`)

          for env in "${envs[@]}"; do
              IFS='=' read -r key value string <<< "$env"
              if [[ "$key" != "" && "$value" != "" ]]; then
                  path="${ENV_DIR}/${key}"
                  echo "--> Writing ${path}..."
                  echo -n "$value" > "$path"
              fi
          done
      command:
        - /usr/local/bin/bash
      image: >-
        docker.io/library/bash:5.1.4@sha256:b208215a4655538be652b2769d82e576bc4d0a2bb132144c060efc5be8c3f5d6
      name: prepare
      resources: {}
      securityContext:
        privileged: true
      volumeMounts:
        - mountPath: /cache
          name: cache
        - mountPath: /layers
          name: layers-dir
        - mountPath: /platform
          name: $(params.PLATFORM_DIR)
    - args:
        - '-app=/workspace/source/$(params.CONTEXT_DIR)'
        - '-cache-dir=/cache'
        - '-uid=$(params.USER_ID)'
        - '-gid=$(params.GROUP_ID)'
        - '-layers=/layers'
        - '-platform=/platform'
        - '-report=/layers/report.toml'
        - '-process-type=$(params.PROCESS_TYPE)'
        - '-skip-restore=$(params.SKIP_RESTORE)'
        - '-previous-image=$(params.APP_IMAGE)'
        - '-run-image=$(params.RUN_IMAGE)'
        - $(params.APP_IMAGE)
      command:
        - /cnb/lifecycle/creator
      image: $(params.BUILDER_IMAGE)
      imagePullPolicy: Always
      name: create
      resources: {}
      securityContext:
        runAsGroup: 1000
        runAsUser: 1000
      volumeMounts:
        - mountPath: /cache
          name: cache
        - mountPath: /layers
          name: layers-dir
        - mountPath: /platform
          name: $(params.PLATFORM_DIR)
    - args:
        - '-c'
        - >
          #!/usr/bin/env bash

          set -e

          cat /layers/report.toml | grep "digest" | cut -d'"' -f2 | cut -d'"'
          -f2 | tr -d '\n' | tee $(results.shp-image-digest.path)
      command:
        - /bin/sh
      image: >-
        docker.io/library/bash:5.1.4@sha256:b208215a4655538be652b2769d82e576bc4d0a2bb132144c060efc5be8c3f5d6
      name: results
      resources: {}
      volumeMounts:
        - mountPath: /layers
          name: layers-dir
  volumes:
    - name: cache
    - name: layers-dir
    - name: $(params.PLATFORM_DIR)
  workspaces:
    - name: source
```

Did I miss anything during the installation? Let me know if you need further information, thanks.

**Comment by tpiperatgod** [comment]
This problem is caused by the issue of dapr version 1.4.0, you can refer to this https://github.com/dapr/dapr/issues/3699

You can choose to upgrade dapr to v1.4.1 or downgrade to v1.3.0 to bypass this issue.

We will also fix this issue in the installation script. Thx.

**Comment by lizzzcai** [comment]
Hi @tpiperatgod , thanks for your reply. I just tested it.  v1.4.1 is still not working but`v1.3.1` is working on my side.

**Comment by yaron2** [comment]
Hi, Dapr maintainer here. We are investigating these webhook failures.

**Comment by benjaminhuo** [comment]
> Hi, Dapr maintainer here. We are investigating these webhook failures.

Thanks for the quick response @yaron2

**Comment by artursouza** [comment]
Dapr maintainer here. We did not reproduce this, although we did identify the root cause for this issue. This problem seems to be related to Dapr 1.4.x + Tekton and not necessarily Dapr .14.x + OpenFunction.

**Comment by tpiperatgod** [comment]
Thanks, @artursouza, as you said, this issue should not necessarily be related to OpenFunction. I have now adjusted the version of the Dapr in OpenFunction to v1.3.1 to skip this issue for now(https://github.com/OpenFunction/OpenFunction/pull/123).

**Comment by lizzzcai** [comment]
Hi @tpiperatgod, the issue was fixed in the Dapr `1.4.2` (https://github.com/dapr/dapr/issues/3709). Thanks for the quick support.

**Comment by tpiperatgod** [comment]
Thank you all :thumbsup: . OpenFunction will hold on to Dapr 1.3.1 for now, and upgrade it to 1.4.2 later.
